workflows/bcellmagic.nf

```diff
@@ -46,7 +46,7 @@ if (params.protocol == "pcr_umi"){
 
 // Validate UMI position
 if (params.index_file & params.umi_position == 'R2') {exit 1, "Please do not set `--umi_position` option if index file with UMIs is provided."}
-if (params.umi_length == 0) {exit 1, "Please provide the UMI barcode length in the option `--umi_length`."}
+if (params.umi_length == null) {exit 1, "Please provide the UMI barcode length in the option `--umi_length`. To run without UMIs, set umi_length to 0."}
```

Comment:
I've swapped this back to check for `params.umi_length == null`. Apparently in groovy `!0 == true` so running a no-UMI pipeline would otherwise always trigger the exit condition

Reply:
Hmm it doesn't look like the linting likes us defining an integer param with `null` as the default in the schema. I've set default `umi_length = -1` as a workaround but let me know if you have a different preferred solution.

Reply:
ok weird, I've tested it in the nextflow console (`nextflow console`) and it was interpreted as `false`:
```
def foo = null
var= "hello"
if (!foo) {
  print var
}
```

But yes you are right, the schema does not like an integer defined by default as `null`.  The proposed `-1` solution sounds good to me!

Reply:
The issue was that with no-UMI, we are defining `foo = 0`, which is interpreted the same as `foo = null` and triggers the exit. Cool will stick with the -1 default workaround :-)